### PR TITLE
add leaflet.cluster

### PR DESCRIPTION
**Pull request by PeterTheOne**
Fixes: #141

Seems to work fine. Cluster icons are still missing, should we use the same ones as googlemaps3?

`spiderfy` and other options are not used for now.

TODO before merge:

- [x] add more options, including `spiderfy`
- [x] fix i18n
- [x] add icon
- [x] remove dontAddMarker
- [x] icon: show number of markers per cluster
- [x] fix conflict
- [x] test removing, recreating or refreshing cluster

### CODE REVIEW

File: includes/services/Leaflet/jquery.leaflet.js [hunk 1]
```diff
@@ -25,7 +25,9 @@
 		* @param {Object} markerData Contains the fields lat, lon, title, text and icon
 		* @return {L.Marker}
 		*/
-		this.addMarker = function (properties) {
+		this.addMarker = function (properties, dontAddToMap) {
```
Comment (JeroenDeDauw) [inline includes/services/Leaflet/jquery.leaflet.js]: `dontAddToMap`? what? :) Is this some debug thing?

Comment (PeterTheOne) [inline includes/services/Leaflet/jquery.leaflet.js]: See this https://github.com/JeroenDeDauw/Maps/pull/194/files#diff-4a1096507c3fea5c0fb17f1231c4a617R179 vs. this
https://github.com/JeroenDeDauw/Maps/pull/194/files#diff-4a1096507c3fea5c0fb17f1231c4a617R184 .

For the cluster the markers are not directly added to the map but to the cluster thing. I could also rename it to `addToMap` if you want. ;)

**Comment by PeterTheOne** [comment]
@kghbln I renamed `maps-googlemaps3-par-markercluster` in `en.json` and `qqq.json`, should I have renamed them in all the other languages also?

**Comment by PeterTheOne** [comment]
Options comparison gm3 vs. leaflet:

- maps-googlemaps3-par-clustergridsize (default: 60) -> maxClusterRadius (default: 80)?
- maps-googlemaps3-par-clustermaxzoom -> disableClusteringAtZoom
- maps-googlemaps3-par-clusterzoomonclick -> zoomToBoundsOnClick?
- maps-googlemaps3-par-clusteraveragecenter -> n/a
- maps-googlemaps3-par-clusterminsize -> n/a
- n/a -> spiderfyOnMaxZoom (enabled by default)

**Comment by kghbln** [comment]
> I renamed maps-googlemaps3-par-markercluster in en.json and qqq.json, should I have renamed them in all the other languages also?

I would do so since these messages do not need to be translated again in this case. Depending on the language it can take quite some time until a translation is available since there are sometimes only occasional translators around.

File: includes/services/Leaflet/jquery.leaflet.js (hunk 1 above)
Comment (JeroenDeDauw) [inline includes/services/Leaflet/jquery.leaflet.js]: Right...

* addToMap would be nicer, esp than `!dontAddToMap`
* functions ought to do a single thing (on a given level of abstraction). If they take a boolean flag, you know they fail at this

Comment (PeterTheOne) [inline includes/services/Leaflet/jquery.leaflet.js]: fixed in bc6b2ef

**Comment by PeterTheOne** [comment]
ready for merge

**Comment by JeroenDeDauw** [comment]
Awesome \o/

**Pull request merged**